scripts/conversion_script.py

```diff
@@ -649,13 +653,15 @@ def main():
                     # (if repo existed, the .backup file will remain on system)
                     c2r_repo.keep = True
 
-            if not output.report:
+            if not output.report and returncode != 0:
                 # Try to attach the textual report in the report if we have
                 # json report, otherwise, we would overwrite the report raised
                 # by the exception.
                 output.report = gather_textual_report()
 
-            output.entries = transform_raw_data(data)
+            # Only add entries (report_json) if the returncode is not 0.
+            if returncode != 0:
+                output.entries = transform_raw_data(data)
```

Comment:
Kinda had hard time processing the logic, but it works nicely! Maybe would slightly rephrase it.

```
            conversion_successful = returncode == 0
            if not output.report and not conversion_successful:
                # Attach textual report('report' in output) if one doesn't exists from raised exceptions
                # Only attach it for unsuccessful conversions
                output.report = gather_textual_report()

            # Only add entries ('report_json' in output) if the returncode is not 0.
            if not conversion_successful:
                output.entries = transform_raw_data(data)
```

Reply:
okay, updated.